### DISCUSSION TITLE
Sub collections missing

### DISCUSSION THREAD

**Discussion by auroraanna** [pocketbase/pocketbase#1152] [Q&A]
Pocketbase has no sub collection feature, so a collection inside of a collection or collections that can be created without admin permissions. For my app I need several collections though or alternatively a two collections with the the first one grouping records from the other one via relations. I tried to implement that but it makes everything more complicated. Fetching a set of related record works with the expand parameter but now I can't subscribe to a collection of those related records. I have to subscribe to every single one. And when all records get updated, I get so many responses from the database telling me about the updates. They all then trigger the fetching one all the related records every update. This would be much easier if you could subscribe to those related records by having them in a sub collection which has to be able to be created without admin permission. Am I doing it wrong and are relations just fine to work with or does pocketbase need sub collections?

**Comment by ganigeorgiev** [comment]
I'm not sure what do you mean by "_sub collection which has to be able to be created without admin permission_".

"Sub/inline/nested/etc. collections" is in my todo to research it but there are a lot of ambiguities around the implementation and for now this remains out of the scope for v1 (eg. will they have their own collection rules or will share the ones from the parent collection; how data will be submitted with the nested collection record(s); how uploading files and multipart/form-data requests in general will work in this case, etc.).

Realtime filters support will be added in the near future (most likely somewhere after the filter enhancements, JS server-side event hooks and the private/protected files; once I have a more clear idea of the implementation, I'll create an issue and will add it to the [roadmap](https://github.com/orgs/pocketbase/projects/2)).

**Comment by auroraanna** [comment]
So, I am not sure if something like sub collections is really needed since I managed to solve my problem with more collections and relations.
